ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDTFStack.java

```diff
@@ -63,13 +64,21 @@ public StructObjectInspector initialize(ObjectInspector[] args)
     }
     if (!(args[0] instanceof ConstantObjectInspector)) {
       throw new UDFArgumentException(
+          "The first argument to STACK() must be a constant.");
+    }
+    final Object value = ((ConstantObjectInspector) args[0]).getWritableConstantValue();
+    if (value == null) {
+      throw new UDFArgumentException("The first argument of STACK() must not be null.");
+    }
+    if (!(value instanceof IntWritable)) {
```

Comment:
Can a long be passed here (LongWritable)?

Reply:
Yes, but I believe there is no valid use case for two reasons.
- This UDF can't process N > Integer.MAX_VALUE anyway, as the maximum length of Java's arrays is Integer.MAX_VALUE
- I don't think anyone prefers to write `stack(cast(3 as bigint), a, b, c, ...`